src/alivedetection.c

```diff
@@ -1254,6 +1254,88 @@ send_arp (__attribute__ ((unused)) gpointer key, gpointer value,
     }
 }
 
+/**
+ * @brief Send all dead hosts to ospd-openvas.
+ *
+ * All hosts which are not identified as alive are sent to ospd-openvas. This is
+ * needed for the calculation of the progress bar for gsa in ospd-openvas.
+ *
+ * @return 0 on success, <0 on failure.
+ */
+static void
+send_dead_hosts_to_ospd_openvas (void)
+{
+  kb_t main_kb = NULL;
+  int maindbid;
+  char buf[256];
+  int hosts_in_chunk = 0;
+
+  GHashTableIter target_hosts_iter;
+  gpointer host_str, value;
+  GString *chunked_hosts;
+
+  maindbid = atoi (prefs_get ("ov_maindbid"));
+  main_kb = kb_direct_conn (prefs_get ("db_address"), maindbid);
+
+  if (!main_kb)
+    {
+      g_warning ("%s: Could not connect to main_kb.", __func__);
+      return;
+    }
+
+  chunked_hosts = g_string_new (NULL);
+  for (g_hash_table_iter_init (&target_hosts_iter, hosts_data.targethosts);
+       g_hash_table_iter_next (&target_hosts_iter, &host_str, &value);)
+    {
+      /* If a host in the target hosts is not in the list of alive hosts we know
+       * it is dead and we have to send it to ospd-openvas */
+      if (!g_hash_table_contains (hosts_data.alivehosts, host_str))
+        {
+          g_string_append (chunked_hosts, host_str);
+          hosts_in_chunk++;
+
+          if (hosts_in_chunk == 1000)
+            {
+              /* Add start and end of message. */
+              g_string_prepend (chunked_hosts, "DEADHOST||| ||| |||");
+              g_string_append (chunked_hosts, "|||Host dead");
```

Comment:
The right message here should be following. This allows ospd-openvas insert the host detail log.
```suggestion
              g_string_prepend (chunked_hosts, "DEADHOST||| |||general/Host_Details|||");
              g_string_append (chunked_hosts, "|||<host><detail><name>Host dead</name><value>1</value>source><description/><type/><name/></source></detail></host>");
```